src/lib/utils/debounce.ts

```diff
@@ -0,0 +1,39 @@
+// Debounce the function call to the trailing edge.
+export function debounce(func: Function, wait: number): Function {
+  let lastTimeout;
+  return function(...args) {
+    // @ts-ignore-next-line
+    const that = this;
+    return new Promise<void>(resolve => {
+      clearTimeout(lastTimeout);
+      lastTimeout = setTimeout(() => {
+        lastTimeout = null;
+        func.apply(that, args);
```

Comment:
As you are using arrow functions, I think you can get rid of the `const that = this` logic if you spread the args. wdyt @Haroenv @eunjae-lee 
```suggestion
        func(...args);
```

Reply:
It works in some cases that way.
https://codesandbox.io/s/serene-shape-k7upd?file=/src/index.js
If you see the examples in the link above, the example 2 won't work without `that`.
Of course, it's not that important in our case, though.

Reply:
Oh ok! I didn't know, but isn't this issue related to how we create the object and not how `debounce` works? I know it's not the same but if we use `b.data` in the example 2, it will work

Reply:
since our functions don't use `this`, I think it might not even be needed though

Reply:
okay, since it's not used yet, let's remove it. https://github.com/algolia/instantsearch.js/pull/4581/commits/81cba95bccc361888319ce1fa38b079e953dbf1a